akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -106,6 +107,9 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
   if (persistenceId eq null)
     throw new IllegalArgumentException("persistenceId must not be null")
 
+  // Don't use it directly, but instead call internalLogger() (see below)
+  private val logger = LoggerFactory.getLogger(this.getClass)
```

Comment:
Internal logger is now `akka.persistence.typed.internal.EventSourcedBehaviorImpl` and ensure MDC is set each time we use it inside EventSourcedBehaviorImpl, but also on every call to `BehaviorSetup. internalLogger `

Reply:
We lose the capability of getting/enabling those logs for only one user-type of EventSourcedBehavior, but maybe that's ok.

Reply:
yes, that's indeed the main drawback. 

I think that's ok, since this is more a debug level thing. Users can still define a logger filter and use the MDC information to of filter out some message, for instance using the `persistentId`. That would be possible if they are using the id type hint for instance. 

If we find it a major issue, we can append the behavior name to the log. Will feel a bit hacky though.

Reply:
Sounds ok to me with that drawback for now.